MdeModulePkg/Universal/SmbiosDxe/SmbiosDxe.c

```diff
@@ -1408,6 +1408,242 @@ SmbiosTableConstruction (
   }
 }
 
+/**
+  Validates a SMBIOS 2.0 table entry point.
+
+  @param  SmbiosTable   The SMBIOS_TABLE_ENTRY_POINT to validate.
+
+  @retval TRUE           SMBIOS table entry point is valid.
+  @retval FALSE          SMBIOS table entry point is malformed.
+
+**/
+STATIC
+BOOLEAN
+ValidateSmbios20Table(
+  IN SMBIOS_TABLE_ENTRY_POINT      *SmbiosTable
+) {
+  UINT8 Checksum;
+
+  if (CompareMem (SmbiosTable->AnchorString, "_SM_", 4) != 0) {
+    return FALSE;
+  }
+  if (SmbiosTable->EntryPointLength < 0x1E) {
+    return FALSE;
+  }
+  if (SmbiosTable->MajorVersion < 2) {
+    return FALSE;
+  }
+  if (SmbiosTable->SmbiosBcdRevision > 0 &&
+    (SmbiosTable->SmbiosBcdRevision >> 4) < 2) {
+    return FALSE;
+  }
+  if (SmbiosTable->TableLength == 0) {
+    return FALSE;
+  }
+  if (SmbiosTable->TableAddress == 0 ||
+    SmbiosTable->TableAddress == ~0) {
+    return FALSE;
+  }
+
+  Checksum = CalculateSum8((UINT8 *) SmbiosTable,
+    SmbiosTable->EntryPointLength);
+  if (Checksum != 0) {
+    return FALSE;
+  }
+
+  Checksum = CalculateSum8((UINT8 *) SmbiosTable + 0x10,
+    SmbiosTable->EntryPointLength - 0x10);

```

Comment:
What does this checksum calculation mean? If there is a reason, please add comments.

Reply:
It is already changed in new pull request